### PR TITLE
Fixes #19892: importing directives from rudder 6.2 breaks everything in 7.0

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/19892

### CODE REVIEW

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/eventlog/EventLogFactory.scala [hunk 1]
```diff
@@ -522,7 +522,7 @@ class EventLogFactoryImpl(
           <displayName>{modifyDiff.name}</displayName>{
             modifyDiff.modName.map(x => SimpleDiff.stringToXml(<name/>, x) ) ++
             modifyDiff.modTechniqueVersion.map(x => SimpleDiff.toXml[TechniqueVersion](<techniqueVersion/>, x){ v =>
-              Text(v.toString)
+              Text(v.serialize)
```
Comment (fanf) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/eventlog/EventLogFactory.scala:525]: this one was putting `TechniqueVersion(....)` in xml

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/marshalling/XmlSerialisationImpl.scala [hunk 2]
```diff
@@ -201,7 +201,7 @@ class DirectiveSerialisationImpl(xmlVersion:String) extends DirectiveSerialisati
           <id>{directive.id.uid.value}</id>
       ::  <displayName>{directive.name}</displayName>
       ::  <techniqueName>{ptName.value}</techniqueName>
-      ::  <techniqueVersion>{directive.techniqueVersion}</techniqueVersion>
+      ::  <techniqueVersion>{directive.techniqueVersion.serialize}</techniqueVersion>
```
Comment (fanf) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/marshalling/XmlSerialisationImpl.scala:204]: oups, it was this on in config-repo xml, the previous was in event logs

**Comment by fanf** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/marshalling/XmlSerialisationImpl.scala [hunk 3]
```diff
@@ -175,7 +175,7 @@ class ActiveTechniqueSerialisationImpl(xmlVersion:String) extends ActiveTechniqu
   def serialise(activeTechnique:ActiveTechnique):  Elem = {
     createTrimedElem(XML_TAG_ACTIVE_TECHNIQUE, xmlVersion) (
         <id>{activeTechnique.id.value}</id>
-        <techniqueName>{activeTechnique.techniqueName}</techniqueName>
+        <techniqueName>{activeTechnique.techniqueName.value}</techniqueName>
```
Comment (fanf) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/marshalling/XmlSerialisationImpl.scala:178]: this one was the likely cause of all the `TechniqueName()` in the technique tree after a reload archive from master

**Review by ncharles — APPROVED** [review]

**Pull request merged**